app/controllers/api/v1/PostsController.rb

```diff
@@ -0,0 +1,95 @@
+class Api::V1::PostsController < ApplicationController
+  include Authentication
+
+  def index
+    posts = Post.all
+
+    if posts
+      render json: {data: posts}, status: :ok
+    else
+      render json: posts.errors, status: :bad_request
+    end
+  end
+
+  def create
+    if authenticate
+      user = User.find_by(username: request.headers[:username])
+      post = Post.new(title: post_params[:title], description: post_params[:description], slug: "", comments_count: 0, likes_count: 0, user: user)
```

Comment:
You can set the default values for slug,  comments_count and likes_count instead of setting them here for each new post.
Then you can use post_params directly as: `Post.new(post_params)`
Append user in post_params itself.

Reply:
appended by setting it in default like this

`params.permit(:title, :description).with_defaults(slug: "", likes_count: 0, comments_count: 0, user: @user)`
@user is value being fetched from the application_controller.